Jenkinsfile

```diff
@@ -219,28 +217,70 @@ pipeline {
+                  dir("${BASE_DIR}") {
+                    sh './.ci/scripts/benchmarks.sh'
+                  }
+                }
+              }
+              post {
+                always {
+                  archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/${env.REPORT_FILE}", onlyIfSuccessful: false)
+                  catchError(message: 'sendBenchmarks failed', buildResult: 'FAILURE') {
+                    log(level: 'INFO', text: "sendBenchmarks is ${env.CHANGE_ID?.trim() ? 'not enabled for PRs' : 'enabled for branches'}")
+                    whenTrue(env.CHANGE_ID == null){
+                      sendBenchmarks(file: "${BASE_DIR}/${env.REPORT_FILE}", index: 'benchmark-rum-js')
+                    }
+                  }
+                  catchError(message: 'deleteDir failed', buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
+                    deleteDir()
+                  }
+                }
               }
             }
-          }
-          post {
-            always {
-              archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/${env.REPORT_FILE}", onlyIfSuccessful: false)
-              catchError(message: 'sendBenchmarks failed', buildResult: 'FAILURE') {
-                log(level: 'INFO', text: "sendBenchmarks is ${env.CHANGE_ID?.trim() ? 'not enabled for PRs' : 'enabled for branches'}")
-                whenTrue(env.CHANGE_ID == null){
-                  sendBenchmarks(file: "${BASE_DIR}/${env.REPORT_FILE}", index: 'benchmark-rum-js')
+            stage('Load testing') {
+              agent { label 'metal' }
+              options {
+                warnError('load testing failed')
+              }
+              environment {
+                REPORT_FILE = 'apm-agent-load-testing-results.json'
+              }
+              steps {
+                withGithubNotify(context: 'Load testing') {
+                  deleteDir()
+                  unstash 'source'
+                  unstash 'cache'
+                  dockerLogin(secret: "${DOCKER_ELASTIC_SECRET}", registry: 'docker.elastic.co')
+                  dir("${BASE_DIR}") {
+                    sh ".ci/scripts/load-testing.sh ${env.STACK_VERSION}"
+                  }
                 }
               }
-              catchError(message: 'deleteDir failed', buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
-                deleteDir()
+              post {
+                always {
+                  archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/${env.REPORT_FILE}", onlyIfSuccessful: false)
+                  catchError(message: 'sendBenchmarks failed', buildResult: 'FAILURE') {
+                    log(level: 'INFO', text: "sendBenchmarks is ${env.CHANGE_ID?.trim() ? 'not enabled for PRs' : 'enabled for branches'}")
+                    whenTrue(env.CHANGE_ID == null){
+                      sendBenchmarks(file: "${BASE_DIR}/${env.REPORT_FILE}", index: 'benchmark-rum-js')
```

Comment:
@jahtalab , It will use the existing index, if a new one is required then a follow up is needed.

Reply:
Different index is better as we are testing the capability of `apm-server` not the RUM agent. I would keep this functionality separate from our regular benchmarking of the agent itself.

Reply:
I agree a separate index is preferable since these have different schemas, but can't we use the `_index` field that is created [in the load test result file](https://github.com/elastic/apm-agent-rum-js/blob/master/scripts/apm-server-load-test.js#L261)?

Reply:
Then we need to update the L277 to reflect the new index to benefit from the `sendBenchmarks` step